feat(pager): add optional page change event

louislewis2 opened the pull request:
add an optional page change event, which will be called when the page value changes. this will simplify use cases where the pager is used stand alone or when server side paging is required. update readme to reflect the bindable property

Commit:
feat(pager): add optional page change event

add an optional page change event, which will be called when the page value changes. this will simplify use cases where the pager is used stand alone or when server side paging is required. update readme to reflect the bindable property

RWOverdijk commented on src/pager.js:
```diff
@@ -38,6 +39,9 @@ export class Pager {
   pageChanged(newValue, oldValue) {
     if (newValue !== oldValue) {
       this.goToPage(newValue);
+	  if (this.onPageChanged !== null) {
```
`if (typeof this.onPageChanged === 'function') {}` is probably a better idea.

Also, add two spaces in front of the if (indentation is wrong now) and add a newline before the if.

But reading your issue, I'm not sure this solves your problem. You might want to wait for it to finish loading data. If not, this PR is fine :)

RWOverdijk reviewed (changes requested):
Almost perfect!

Commit:
fix(pager): replace null check with function check

onPageChanged was being checked for null, it now correctly checks whether it is a function.

RWOverdijk commented:
I'm sorry to bother you, but indentation is still wrong, mainly because you're using tabs (we use spaces). You also left out the typeof check. It should look like this:

```js
  pageChanged(newValue, oldValue) {
    if (newValue !== oldValue) {
      this.goToPage(newValue);

      if (typeof this.onPageChanged === 'function') {
        this.onPageChanged(newValue, oldValue);
      }
    }
  }
```

louislewis2 commented:
Sorry, battling with an editor. currently using notepad to edit, when I open the source in visual studio code, it messes up all the files with styling.. Let me try correct it again ;)

RWOverdijk commented:
No problem :) Thanks for contributing!

Commit:
fix(pager): add missing typeof statement

correct styling for the affected code

louislewis2 commented:
😨 Did it come right this time around? 😣

RWOverdijk commented:
Mmmm no haha. You can just use my snippet of code. I'd do it, but you should have your name on this code :)

louislewis2 commented:
What editor are you using? Maybe I can download the same one and just get it right...

Commit:
fix(pager): correct styling

louislewis2 commented:
Okay, I tried once again..... 😁 
This time I disabled the format on save and turned on view white space.
With a little bit of luck it might be closer to the truth this time.

RWOverdijk commented:
It's not, sorry. You changed the formatting to 4 spaces instead of 2

Commit:
style(pager): try correct spacing count

louislewis2 commented:
If that one is not right, then I don't know.
However I do think that adding in a editor config file for visual studio code can maybe save some future trouble?

RWOverdijk commented:
You did it! :D I'll release this

louislewis2 commented:
Thanks for your patience on this one. It is much appreciated. I'm certain next time will be much smoother ;)

RWOverdijk commented:
Published as 0.2.0

Low version for something this stable makes me happy :)
